tests/test_utils_mean.py

```diff
@@ -64,45 +67,76 @@ def test_riemann_mean(init, get_covmats_params):
 
 def test_euclid_mean(get_covmats):
     """Test the euclidean mean"""
-    n_trials, n_channels = 100, 3
-    covmats = get_covmats(n_trials, n_channels)
+    n_matrices, n_channels = 100, 3
+    covmats = get_covmats(n_matrices, n_channels)
     C = mean_euclid(covmats)
     assert C == approx(covmats.mean(axis=0))
 
 
 def test_identity_mean(get_covmats):
     """Test the identity mean"""
-    n_trials, n_channels = 100, 3
-    covmats = get_covmats(n_trials, n_channels)
+    n_matrices, n_channels = 100, 3
+    covmats = get_covmats(n_matrices, n_channels)
     C = mean_identity(covmats)
     assert np.all(C == np.eye(n_channels))
 
 
 def test_alm_mean(get_covmats):
     """Test the ALM mean"""
-    n_trials, n_channels = 3, 3
-    covmats = get_covmats(n_trials, n_channels)
+    n_matrices, n_channels = 3, 3
+    covmats = get_covmats(n_matrices, n_channels)
     C_alm = mean_alm(covmats)
     C_riem = mean_riemann(covmats)
     assert C_alm == approx(C_riem)
 
 
 def test_alm_mean_maxiter(get_covmats):
     """Test the ALM mean with max iteration"""
-    n_trials, n_channels = 3, 3
-    covmats = get_covmats(n_trials, n_channels)
+    n_matrices, n_channels = 3, 3
+    covmats = get_covmats(n_matrices, n_channels)
     C = mean_alm(covmats, maxiter=1)
     assert C.shape == (n_channels, n_channels)
 
 
-def test_alm_mean_2trials(get_covmats):
-    """Test the ALM mean with 2 trials"""
-    n_trials, n_channels = 2, 3
-    covmats = get_covmats(n_trials, n_channels)
+def test_alm_mean_2matrices(get_covmats):
+    """Test the ALM mean with 2 matrices"""
+    n_matrices, n_channels = 2, 3
+    covmats = get_covmats(n_matrices, n_channels)
     C = mean_alm(covmats)
     assert np.all(C == geodesic_riemann(covmats[0], covmats[1], alpha=0.5))
 
 
+@pytest.mark.parametrize("init", [True, False])
+def test_riemann_mean_masked_shape(init, get_covmats, get_masks):
+    """Test the masked riemann mean"""
+    n_matrices, n_channels = 10, 6
+    covmats = get_covmats(n_matrices, n_channels)
+    masks = get_masks(n_matrices, n_channels)
+    if init:
+        C = maskedmean_riemann(covmats, masks, init=covmats[0])
+    else:
+        C = maskedmean_riemann(covmats, masks)
+    assert C.shape == (n_channels, n_channels)
+
+
+@pytest.mark.parametrize("init", [True, False])
+def test_riemann_mean_nan_shape(init, get_covmats):
+    """Test the riemann nan mean"""
+    n_matrices, n_channels = 10, 6
+    covmats = get_covmats(n_matrices, n_channels)
+    emean = np.mean(covmats, axis=0)
+    for i in range(n_matrices):
+        corrup_channels = np.random.randint(n_channels, size=n_channels // 2)
```

Comment:
@sylvchev , how can we use the reproducible random seed here? because it seems not explicitely available...

Reply:
there's a fixture for that ©️  I'll push the modification.